README.md

```diff
@@ -1,107 +1,115 @@
+* Define clear JNI interfaces for native libraries
+* Switch the native backends easily
+* Reuse Spark’s distributed control flow
+* Manage data sharing between JVM and native
+* Extend support to native accelerators
 
-# Issues to Solve and status
+## Architecture
 
-The code is still not completely cleaned now. The work is still WIP.
+The overview chart is like below. Spark physical plan is transformed to substrait plan. Substrait is to create a well defined cross-language specification for data compute operations. More details can be found from https://substrait.io/. Then substrait plan is passed to native through JNI call. In native the operator chain should be built and start to run. We use Spark3.0's columnar API as the data interface, so the native library should return Columnar Batch to Spark. We may need to wrap the columnar batch for each native backend. Gazelle engine's c++ code use Apache Arrow data format as its basic data format, so the returned data to Spark JVM is ArrowColumnarBatch.
+<p align="center">
+<img src="https://user-images.githubusercontent.com/47296334/199617207-1140698a-4d53-462d-9bc7-303d14be060b.png" width="800">
+</p>
+There are several native libraries we may offload. Currently we are working on Clickhouse and Velox as native backend. Velox is a C++ database acceleration library which provides reusable, extensible, and high-performance data processing components. More details can be found from https://github.com/facebookincubator/velox/. We also implemented a basic backend using Arrow Computer Engine which is for reference only. Gluten can also be easily extended to any accelerator libraries as backend.
 
-Not all the operators and functions are added. 22 queries of TPCH can pass. We are working on TPCDS queries. <b> We haven't pass Spark UTs, the work is just started and expected to finish in Q4 2022. So any other queries except H/DS may run into issues. Feel free to submit it to our issue list.</b>
+There are several key component in GLuten:
+* Query plan conversion which convert Spark's physical plan into substrait plan in each stage.
+* Unified memory management in Spark is used to control the native memory allocation as well
+* Columnar shuffle is used to shuffle columnar data directly. The shuffle service still reuse the one in Spark core. The exchange operator is reimplemented to support columnar data format
+* For unsupported operators or functions Gluten fallback the operator to Vanilla Spark. There are C2R and R2C converter to convert the columnar data and Spark's internal row data. Both C2R and R2C are implemented natively as well
+* Metrics are very important to get insight of Spark's execution, identify the issues or bottlenecks. Gluten collects the metrics from native library and shows in Spark UI.
+* Shim layer is used to support mutiple releases of Spark. Gluten does not and will not support all the Spark releases. The plan is to support the latest 2-3 spark stable release only.
```

Comment:
Gluten does not and will not support all the Spark releases. The plan is to support the latest 2-3 spark stable release only.

>> Gluten only plans to support the latest 2-3 spark stable releases, with no plans to add support on older spark releases. Current support is on Spark 3.2 and Spark 3.3.